docs/source/user_guide/user_guide/getting_started.md

```diff
@@ -0,0 +1,232 @@
+
+ti.sync()
+start = time.time()
+
+lcg_ti(B, 1000, a)
+
+ti.sync()
+end = time.time()
+print("elapsed", end - start)
+```
+
+When run on a Macbook Air M4, the output is something like:
+```
+# [GsTaichi] version 1.8.0, llvm 15.0.7, commit 5afed1c9, osx, python 3.10.16
+# [GsTaichi] Starting on arch=metal
+# elapsed 0.04660296440124512
+```
+Around 120x faster.
+
+On one of our linux boxes with a 5090 GPU, the results are:
+- numpy: 6.90 seconds
+- gstaichi: 0.0199 seconds
+- => 346 times faster
+
+## What does GsTaichi do with the kernel function?
+
+- any top level for loops are parallelized across the GPU cores (or CPU, if you run on CPU)
+    - in our case, there will be 16,000 threads
+    - compared to just a single thread in the numpy case
+
+## fields: even faster
+
+GsTaichi ndarrays are easy to use, and flexible, but we can increase speed by another ~30% or so (depending on the kernel), by using fields.
+
+The kernel above doesn't load or store data except at the start and end: it's just exercising the GPU APU. To see the difference between Taichi ndarray and GsTaichi field runtime speed, we need a kernel that does more loads and stores.
+
+We'll do a simple kernel that copies from one tensor to another. To avoid simply measuring the latency to read and write from/to global memory, we'll read and write the same values repeatedly.
+
+```
+import argparse
+import time
+import gstaichi as ti
+
+ti.init(arch=ti.gpu)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--use-field", action="store_true")
+args = parser.parse_args()
+
+use_field = args.use_field
+if use_field:
+    V = ti.field
+    ParamType = ti.Template
+else:
+    V = ti.ndarray
+    ParamType = ti.types.NDArray[ti.i32, 1]
+
+@ti.kernel
+def copy_memory(N: int, a: ParamType, b: ParamType) -> None:
+    for n in range(N):
+        b[n % 100] = a[n % 100]
+
+N = 20_000
+a = V(ti.i32, (100,))
+b = V(ti.i32, (100,))
+
+# warmup
+copy_memory(N, a, b)
+
+num_its = 1000
+
+ti.sync()
+start = time.time()
+for it in range(num_its):
+    copy_memory(N, a, b)
+ti.sync()
+end = time.time()
+iteration_time = (end - start) / num_its * 1_000_000
+print("iteration time", iteration_time, "us")
+```
+
+Here are the outputs, using a 5090 gpu, on ubuntu:
+```
+$ python doc/mem_copy.py
+[GsTaichi] version 1.8.0, llvm 15.0.4, commit b4755383, linux, python 3.10.15
+[GsTaichi] Starting on arch=cuda
+iteration time 29.45709228515625 us
+
+$ python doc/mem_copy.py --use-field
+[GsTaichi] version 1.8.0, llvm 15.0.4, commit b4755383, linux, python 3.10.15
+[GsTaichi] Starting on arch=cuda
+iteration time 21.44002914428711 us
+```
+=> in this test, fields are around 28% faster than ndarrays.
+
+(The exact ratio obviously depends on the kernel. In addition it's possible to construct toy examples like this where ndarray appears to be faster than fields, but in actual kernels, such as Genesis func narrow phase kernel, for collisions, we observe fields are around ~30% faster)
```

Comment:
"but in actual kernels" -> "for kernels used in practice applications" or "in production" ?

Because what means being an "actual kernel" is not very clear. Poor `copy_memory` kernel, it is not your fault it you are born like this :(